### PR TITLE
Add DataAccess Section to details page

### PR DESCRIPTION
**Pull request by BronzBierd**
# Fixes or Implements [AB#15411](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/15411)

### Expand Patient Details page

1. Added Data Access Section component that allows for toggling data source access for patient
2. Added Added AgentAuditHistory to display all of the agent's reasons for making changes to a patients account. (All changes performed against the Hdid.

### Add AgentAudit Components
1. AuditReasonDialog - Acts as a confirmation
3. AgentAuditHistory - Displays reasons for changes in a Expansion panel (I can think of a few improvements here but can be done as an improvement ticket)

## Testing

- [ ] Unit Tests Updated
- [ ] Functional Tests Updated
- [X] Not Required

### Patient Details Page
image

### Dataset Access With Changes
image

### Audit Reason Dialog
image

### Loaders
image

### Confirmation snackbar
image

### Agent Audit History
image

## Notes
Functional tests will fail due to removing the confirmation dialog from the Protect/Unprotect workflows.

## Items to Review:

-   [General PR Guidelines](https://github.com/bcgov/healthgateway/wiki/PRguidance)

### CODE REVIEW

File: Apps/Admin/Client/Components/AgentAudit/AgentAuditHistory.razor [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+                      data-testid="agent-audit-history-table">
+                <RowTemplate>
+                    <MudTd data-testid="agent-action-history-item" DataLabel="Agent-History">
+                        <MudListItem Class="background-secondary">
+                            <MudText Typo="Typo.h6" class="mb-2">
+                                <b>@context.AgentUsername</b>
+                            </MudText>
+                            <MudText Typo="Typo.body2">@context.Reason</MudText>
+                            <MudText Typo="Typo.caption">
+                                <b>@context.TransactionDateTime</b>
```
Comment (BrianMaki) [inline Apps/Admin/Client/Components/AgentAudit/AgentAuditHistory.razor:27]: Is Transaction Date Time being displayed in PST?  If not, should it?

Comment (BronzBierd) [reply Apps/Admin/Client/Components/AgentAudit/AgentAuditHistory.razor:27]: I believe it should, thanks this is something I missed.

**Review by BrianMaki — APPROVED** [review]

File: Apps/Admin/Client/Store/BaseAgentAuditAction.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+// -------------------------------------------------------------------------
+namespace HealthGateway.Admin.Client.Store
+{
+    /// <summary>
+    /// The base class for a auditable action that requires a reason within the action.
```
Comment (MikeLyttle) [inline Apps/Admin/Client/Store/BaseAgentAuditAction.cs:19]:
```suggestion
    /// The base class for an auditable action that requires a reason.
```

File: Apps/Admin/Client/Components/Support/DatasetAccessSection.razor.cs [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+    using HealthGateway.Common.Data.Constants;
+    using Microsoft.AspNetCore.Components;
+    using MudBlazor;
+
+    /// <summary>
+    /// Backing logic for support's dataset access section.
+    /// </summary>
+    public partial class DatasetAccessSection : FluxorComponent
+    {
+        private readonly IEnumerable<DataSource> dataSources = Enum.GetValues(typeof(DataSource)).Cast<DataSource>().Where(x => x != DataSource.Unknown);
```
Comment (MikeLyttle) [inline Apps/Admin/Client/Components/Support/DatasetAccessSection.razor.cs:35]:
```suggestion
        private readonly IEnumerable<DataSource> dataSources = Enum.GetValues<DataSource>().Where(x => x != DataSource.Unknown);
```

File: Apps/Admin/Client/Components/Support/DatasetAccessSection.razor.cs [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+                .ConfigureAwait(true);
+
+            DialogResult result = await dialog.Result.ConfigureAwait(true);
+            if (!result.Canceled)
+            {
+                this.Snackbar.Add("Patient's dataset access has been updated.", Severity.Success);
+            }
+        }
+
+
```
Comment (MikeLyttle) [inline Apps/Admin/Client/Components/Support/DatasetAccessSection.razor.cs:152]:
```suggestion

```

File: Apps/Admin/Client/Components/Support/DatasetAccessSection.razor.cs [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+                    options)
+                .ConfigureAwait(true);
+
+            DialogResult result = await dialog.Result.ConfigureAwait(true);
+            if (!result.Canceled)
+            {
+                this.Snackbar.Add("Patient's dataset access has been updated.", Severity.Success);
+            }
+        }
+        private bool IsBlocked(DataSource dataSource)
```
Comment (MikeLyttle) [inline Apps/Admin/Client/Components/Support/DatasetAccessSection.razor.cs:151]:
```suggestion
        }
        
        private bool IsBlocked(DataSource dataSource)
```

**Review by MikeLyttle — APPROVED** [review]

**Pull request merged**